pkg/grizzly/providers.go

```diff
@@ -0,0 +1,146 @@
+package grizzly
+
+import "fmt"
+
+// Resource represents a single Resource destined for a single endpoint
+type Resource struct {
+	UID      string      `json:"uid"`
+	Filename string      `json:"filename"`
+	Handler  Handler     `json:"handler"`
+	Detail   interface{} `json:"detail"`
+	Path     string      `json:"path"`
+}
+
+// Kind returns the 'kind' of the resource, i.e. the type of the provider
+func (r *Resource) Kind() string {
+	return r.Handler.GetName()
+}
+
+// Key returns a key that combines kind and uid
+func (r *Resource) Key() string {
+	return fmt.Sprintf("%s/%s", r.Kind(), r.UID)
+}
+
+// GetRepresentation Gets the string representation for this resource
+func (r *Resource) GetRepresentation() (string, error) {
```

Comment:
Yes. This actually makes sense. Let me do this throughout.

Reply:
Not so simple. `String()` should return just `string`, but because this will invoke yaml/json marshal, there's gonna be an error that I cannot otherwise know what to do with.

Reply:
`Error: could not manifest JSON into String` ?